model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/TransferTests.xtend

```diff
@@ -576,4 +576,210 @@ class TransferTests {
+            model Test;
+
+            import judo::types;
+
+            transfer T {
+            	initializer;
+            }
+        '''.parse => [
+            assertError(JsldslPackage::eINSTANCE.transferInitializerDeclaration, JslDslValidator.INVALID_DECLARATION, "Initializer is not allowed in unmapped transfer object.")
+        ]
+    }
+
+    @Test
+    def void testTransferInitializerDuplicated() {
+        '''
+			model Test;
+			
+			import judo::types;
+			
+			entity A {}
+			
+			transfer T(A a) {
+				initializer;
+				initializer;
+			}
+        '''.parse => [
+            assertError(JsldslPackage::eINSTANCE.transferInitializerDeclaration, JslDslValidator.DUPLICATE_INITIALIZER)
+        ]
+    }
+
+    @Test
+    def void testTransferCreateUnmapped() {
+        '''
+            model Test;
+
+            import judo::types;
+
+            transfer T {
+            	create;
+            }
+        '''.parse => [
+            assertError(JsldslPackage::eINSTANCE.transferCreateDeclaration, JslDslValidator.INVALID_DECLARATION, "Create is not allowed in unmapped transfer object.")
+        ]
+    }
+
+    @Test
+    def void testTransferCreateDuplicated() {
+        '''
+			model Test;
+			
+			import judo::types;
+			
+			entity A {}
+			
+			transfer T(A a) {
+				create;
+				create;
+			}
+        '''.parse => [
+            assertError(JsldslPackage::eINSTANCE.transferCreateDeclaration, JslDslValidator.DUPLICATE_CREATE)
+        ]
+    }
+
+    @Test
+    def void testTransferCreateIncompatibleParameter() {
+        '''
+			model Test;
+			
+			import judo::types;
+			
+			entity A {}
+			entity B {}
+			
+			transfer TA(A a) {
+				create(TB);
+			}
+			
+			transfer TB (B b) {}
+        '''.parse => [
+            assertError(JsldslPackage::eINSTANCE.transferCreateDeclaration, JslDslValidator.INVALID_DECLARATION, "Create parameter must be compatible to transfer object type.")
+        ]
+    }
+
+    @Test
+    def void testTransferCreateOk() {
+        '''
+			model Test;
+			
+			import judo::types;
+			
+			entity A {}
+			entity B extends A {}
+			
+			transfer TA(A a) {
+				create(TB);
+			}
+			
+			transfer TB (B b) {}
+        '''.parse => [
+            assertNoErrors
+        ]
+    }
+
+    @Test
+    def void testTransferUpdateUnmapped() {
+        '''
+            model Test;
+
+            import judo::types;
+
+            transfer T {
+            	update;
+            }
+        '''.parse => [
+            assertError(JsldslPackage::eINSTANCE.transferUpdateDeclaration, JslDslValidator.INVALID_DECLARATION, "Update is not allowed in unmapped transfer object.")
+        ]
+    }
+
+    @Test
+    def void testTransferUpdateDuplicated() {
+        '''
+			model Test;
+			
+			import judo::types;
+			
+			entity A {}
+			
+			transfer T(A a) {
+				update;
+				update;
+			}
+        '''.parse => [
+            assertError(JsldslPackage::eINSTANCE.transferUpdateDeclaration, JslDslValidator.DUPLICATE_UPDATE)
```

Comment:
Isn't there a human readable message for duplicate update?

Reply:
not needed, it has its own error code